File: Programmers/Level1/예산/예산.js
function solution(d, budget) {
    const sortedArr = d.sort((a, b) => a - b);
    let answer = 0;
    for (let i = 0; i < sortedArr.length; i++) {
        budget -= sortedArr[i];
        if (budget < 0) {
            return answer;
        }
        answer += 1;
    }
    return answer;
}

const d = [1, 3, 2, 5, 4];
const budget = 9;

console.log(solution(d, budget));

File: Programmers/Level1/예산/예산.py
def solution(d, budget):
    answer = 0
    rest = budget
    d.sort()
    for price in d:
        rest -= price
        if rest < 0:
            return answer
        answer += 1
    return answer


d = [2, 2, 3, 3]
budget = 10

print(solution(d, budget))
